apps/vaporgui/SliceEventRouter.cpp

```diff
@@ -1,140 +1,33 @@
-	qsapp->setWidget(_appearance);
-	qsapp->setWidgetResizable(true);
-	addTab(qsapp,"Appearance");
-
-	_geometry = new SliceGeometrySubtab(this);
-	QScrollArea *qsgeo = new QScrollArea(this);
-	qsgeo->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	qsgeo->setWidget(_geometry);
-	qsgeo->setWidgetResizable(true);
-	addTab(qsgeo, "Geometry");
-
-	_annotation = new SliceAnnotationSubtab(this);
-	QScrollArea *qsAnnotation = new QScrollArea(this);
-	qsAnnotation->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	qsAnnotation->setWidget(_annotation);
-	qsAnnotation->setWidgetResizable(true);
-	addTab(qsAnnotation, "Annotations");
-
-#if 0	
-	QScrollArea *qsimg = new QScrollArea(this);
-	qsimg->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	_image = new SliceImageGUI(this);
-	qsimg->setWidget(_image);
-	addTab(qsimg,"Image");
-	_image->adjustSize();
-
-	ImageFrame *imageFrame = _image->imageFrame;
-	QGLFormat fmt;
-	fmt.setAlpha(true);
-	fmt.setRgba(true);
-	fmt.setDoubleBuffer(true);
-	fmt.setDirectRendering(true);
-	_glSliceImageWindow = new GLSliceImageWindow(fmt,imageFrame,"gltwoddatawindow",imageFrame);
-	imageFrame->attachRenderWindow(_glSliceImageWindow, this);
-#endif
-}
-
-void SliceEventRouter::GetWebHelp(
-	vector <pair <string, string> > &help
-) const {
-	help.clear();
-
-	help.push_back(make_pair(
-		"Slice Overview",
-		"http://www.vapor.ucar.edu/docs/vapor-gui-help/twoD#SliceOverview"
-	));
-
-	help.push_back(make_pair(
-		"Renderer control",
-		"http://www.vapor.ucar.edu/docs/vapor-how-guide/renderer-instances"
-	));
-
-	help.push_back(make_pair(
-		"Data accuracy control",
-		"http://www.vapor.ucar.edu/docs/vapor-how-guide/refinement-and-lod-control"
-	));
-
-	help.push_back(make_pair(
-		"Slice geometry options",
-		"http://www.vapor.ucar.edu/docs/vapor-gui-help/twoD#SliceGeometry"
-	));
-
-	help.push_back(make_pair(
-		"Slice Appearance settings",
-		"http://www.vapor.ucar.edu/docs/vapor-gui-help/twoD#HelloAppearance"
-	));
-
-}
-
-void SliceEventRouter::_updateTab(){
-
-	// The variable tab updates itself:
-	//
-	_variables->Update(
-		GetActiveDataMgr(),
-		_controlExec->GetParamsMgr(),
-		GetActiveParams()
-	);
-
-	_appearance->Update(
-		GetActiveDataMgr(),
-		_controlExec->GetParamsMgr(),
-		GetActiveParams()
-	);
-	_geometry->Update(
-		_controlExec->GetParamsMgr(),
-		GetActiveDataMgr(),
-		GetActiveParams()
-	);
-	_annotation->Update(
-		_controlExec->GetParamsMgr(),
-		GetActiveDataMgr(),
-		GetActiveParams()
-	);
+    AddSubtab("Variables", new PGroup({
+        new PSection("Variable Selection", {
+            new S::PScalarVariableSelector
+        }),
+        new PFidelitySection
+    }));
+    
+    AddSubtab("Appearance", new PGroup({
+        new PTFEditor,
+        new PSection("Slice", {
+            new POrientationSelector,
+            new PSliceSampleLocationSelector,
+            (new PDoubleSliderEdit(SliceParams::_sampleRateTag, "N Samples"))->SetRange(32, 2000)
```

Comment:
The current version of the GUI doesn't explicitly expose the number of samples to use to the user. That is an implementation detail. Instead the user is presented with a unitless quality control knob. Under the hood, higher quality values result in more samples. I'm curious as to what @sgpearse and @shaomeng think